File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/Team.java
package work.socialhub.kslack.entity;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Team {

    private String id;
    private String name;
    private String domain;
    private String emailDomain;
    private TeamIcon icon;

    private String enterpriseId;
    private String enterpriseName;

    public String getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }

    public String getDomain() {
        return this.domain;
    }

    public String getEmailDomain() {
        return this.emailDomain;
    }

    public TeamIcon getIcon() {
        return this.icon;
    }

    public String getEnterpriseId() {
        return this.enterpriseId;
    }

    public String getEnterpriseName() {
        return this.enterpriseName;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setDomain(String domain) {
        this.domain = domain;
    }

    public void setEmailDomain(String emailDomain) {
        this.emailDomain = emailDomain;
    }

    public void setIcon(TeamIcon icon) {
        this.icon = icon;
    }

    public void setEnterpriseId(String enterpriseId) {
        this.enterpriseId = enterpriseId;
    }

    public void setEnterpriseName(String enterpriseName) {
        this.enterpriseName = enterpriseName;
    }

    public static class Profile {
        private String id;
        private String fieldName;
        private Integer ordering;
        private String label;
        private String hint;
        private String type;
        private List<String> possibleValues;
        private ProfileOptions options;
        @SerializedName("is_hidden")
        private boolean hidden;

        public String getId() {
            return this.id;
        }

        public String getFieldName() {
            return this.fieldName;
        }

        public Integer getOrdering() {
            return this.ordering;
        }

        public String getLabel() {
            return this.label;
        }

        public String getHint() {
            return this.hint;
        }

        public String getType() {
            return this.type;
        }

        public List<String> getPossibleValues() {
            return this.possibleValues;
        }

        public ProfileOptions getOptions() {
            return this.options;
        }

        public boolean isHidden() {
            return this.hidden;
        }

        public void setId(String id) {
            this.id = id;
        }

        public void setFieldName(String fieldName) {
            this.fieldName = fieldName;
        }

        public void setOrdering(Integer ordering) {
            this.ordering = ordering;
        }

        public void setLabel(String label) {
            this.label = label;
        }

        public void setHint(String hint) {
            this.hint = hint;
        }

        public void setType(String type) {
            this.type = type;
        }

        public void setPossibleValues(List<String> possibleValues) {
            this.possibleValues = possibleValues;
        }

        public void setOptions(ProfileOptions options) {
            this.options = options;
        }

        public void setHidden(boolean hidden) {
            this.hidden = hidden;
        }
    }

    public static class ProfileOptions {
        @SerializedName("is_protected")
        private boolean _protected;

        public boolean isProtected() {
            return _protected;
        }

        public void setProtected(boolean isProtected) {
            this._protected = isProtected;
        }

        public boolean is_protected() {
            return this._protected;
        }

        public void set_protected(boolean _protected) {
            this._protected = _protected;
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/team/Team.kt
package work.socialhub.kslack.entity.team

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class Team {

    var id: String? = null
    var name: String? = null
    var domain: String? = null

    @SerialName("email_domain")
    var emailDomain: String? = null

    var icon: TeamIcon? = null
    var url: String? = null


    @SerialName("enterprise_id")
    var enterpriseId: String? = null

    @SerialName("enterprise_name")
    var enterpriseName: String? = null

    @SerialName("enterprise_domain")
    var enterpriseDomain: String? = null


    @SerialName("default_channels")
    var defaultChannels: Array<String>? = null

    @SerialName("is_verified")
    var isVerified: Boolean? = null

    @SerialName("discoverable")
    var discoverable: String? = null // "invite_only"

    @SerialName("avatar_base_url")
    var avatarBaseUrl: String? = null

    @SerialName("lob_sales_home_enabled")
    var lobSalesHomeEnabled: Boolean? = null


    class Profile {
        var id: String? = null
        var fieldName: String? = null
        var ordering: Int? = null
        var label: String? = null
        var hint: String? = null
        var type: String? = null
        var possibleValues: Array<String>? = null
        var options: ProfileOptions? = null

        @SerialName("is_hidden")
        var isHidden: Boolean = false
    }

    class ProfileOptions {
        @SerialName("is_protected")
        var isProtected: Boolean = false

        fun is_protected(): Boolean {
            return this.isProtected
        }

        fun set_protected(_protected: Boolean) {
            this.isProtected = _protected
        }
    }
}
